marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/4/triggers/entity-model-validate-trigger.xqy

```diff
@@ -0,0 +1,86 @@
+xquery version '1.0-ml';
+
+import module namespace es = "http://marklogic.com/entity-services"
+  at "/MarkLogic/entity-services/entity-services.xqy";
+import module namespace hent = "http://marklogic.com/data-hub/hub-entities" at "/data-hub/5/impl/hub-entities.xqy";
+import module namespace tde = "http://marklogic.com/xdmp/tde"
+        at "/MarkLogic/tde.xqy";
+import module namespace trgr = 'http://marklogic.com/xdmp/triggers' at '/MarkLogic/triggers.xqy';
+
+declare variable $ENTITY-MODEL-COLLECTION as xs:string := "http://marklogic.com/entity-services/models";
+
+declare variable $trgr:uri as xs:string external;
+
+let $entity-def := fn:doc($trgr:uri)/object-node()
+(: Create map version so we can update $ref to external links if we have to :)
+let $entity-def-map as map:map := $entity-def
+let $entity-base-uri := $entity-def/info/baseUri
+let $entity-title := $entity-def/info/title
+let $entity-version := $entity-def/info/version
+let $validate-base-uri := try {
+    xdmp:json-validate-node($entity-def/info, xdmp:unquote('{type: "object", properties: {"baseUri":{type:"string", format:"uri"}}, required: ["baseUri"]}'), "strict")
+  } catch ($ex) {
+    if(($ex/error:code eq "XDMP-JSVALIDATEINVFORMAT" or $ex/error:code eq "XDMP-JSVALIDATEMISSING") and fn:contains($ex/error:format-string/text(), "baseUri")) then (
+      fn:error(fn:QName("error","XDMP-JSVALIDATEINVFORMAT"),fn:concat("A valid Base URI is required (e.g. http://example.org/) for entity: ", $entity-title, "."), fn:concat("Error Message: ", $ex/error:format-string/text()))
```

Comment:
Just confirming - I tested this out by deploying an app with an existing entity definition that doesn't conform to the new requirements, and the deployment failed (as expected) with this message:

```
* What went wrong:
Execution failed for task ':mlDeployApp'.
> Local message: failed to apply resource at documents: Internal Server Error. Server Message: A valid Base URI is required (e.g. http://example.org/) for entity: Customer. (XDMP-JSVALIDATEINVFORMAT): Error Message: XDMP-JSVALIDATEINVFORMAT: Invalid node value: Node value is not valid per format uri, found invalid at baseUri:fn:doc("/entities/Customer.entity.json")/info/text("baseUri") using schema "" . See the MarkLogic server error log for further detail.
```

I think a hard failure is good there, this will hopefully help people catch this problem quickly for existing entity definitions.